indico/modules/rb/operations/bookings.py

```diff
@@ -425,7 +425,7 @@ def split_booking(booking, new_booking_data):
             'recurrence_weekdays': booking.recurrence_weekdays,
         }
 
-        booking.modify(old_booking_data, session.user)
+        booking.modify(old_booking_data, session.user, extra_fields=extra_fields)
```

Comment:
I don't think this is correct! When testing I noticed that the "new" part of the split booking did not have any values for your custom fields. Because this call here updates the **past** part of the booking. You do not want to change that!

Easiest way for you to reproduce: create a booking starting on a past day (enable admin override to be able to do that), and ends in a few days. Then edit that button and change the times (which will trigger splitting). Of course when saving that split booking, the new one should have the data from your extra fields.

Instead the extra data should be applied to the new booking that's created below using `Reservation.create_from_data(...)`

Reply:
Hello @ThiefMaster 👋 that is a great catch, thanks a lot! I totally missed that, as I wasn't too familiar with how the book splitting works.

I believe [this](https://github.com/indico/indico/pull/6126/commits/f7c0cec8a2ef73d2e5ad13ede1c00260899369d5) fixes the new splitting not having the extra data. Basically I just added `extra_fields` to `data` argument passed to `Reservation.create_from_data(...)`.
And [this](https://github.com/indico/indico/pull/6126/commits/3b36efa105113c4356b0181a83a9f939b7b4235f) should fix the old bookings being modified - its up to the plugin to handle the `None` value.